src/main/scala/io/opentargets/etl/backend/target/Target.scala

```diff
@@ -269,28 +281,33 @@ object Target extends LazyLogging {
     logger.info("Adding gene essentiality to target dataframe.")
     val lookup_table =
       ensemblIdLookupDF
-        .select(col("ensgId"), col("symbols"))
-        .withColumn("symbol", explode(col("symbols")))
-        .drop("symbols")
-        .orderBy(col("symbol").asc)
+        .select(col("ensgId"), col("name"))
+        .withColumn("approvedTarget", explode(col("name")))
+        .drop("name")
+        .orderBy(col("approvedTarget").asc)
 
     val targetEssentialityWithEnsgId = geneEssentiality
-      .join(lookup_table, lookup_table("symbol") === geneEssentiality("targetSymbol"), "inner")
+      .join(lookup_table,
+            lookup_table("approvedTarget") === geneEssentiality("targetSymbol"),
+            "inner"
+      )
       .drop(lookup_table.columns.filter(_ != "ensgId"): _*)
       .drop("targetSymbol")
 
-    val targetEssentialityGroupById = targetEssentialityWithEnsgId
+    val test01 = targetEssentialityWithEnsgId
```

Comment:
Would you mind refactoring this?
It may not look right

Reply:
It's been applied